reactioncommerce/reaction
# Fix browsing by product tag

Resolves #2626.

This PR fixes a bug in the `marketplace` branch. The cause was that this app was executing a certain callback that does `setState` in a React component that had been unmounted. This caused an error to be thrown in the browser console.

## Test instructions
1. Create a new product.
1. Create a few (new) tags to that product. 
1. Complete the creation of that product (by filling other fields) and publish it.
1. Add one of these tags to the list of main tags in the navbar. (They are highlighted in the screenshot below.)
1. Then, add two or more tags as sub-tags of the first.
1. Navigate to the root of the app e.g `localhost:3000`.
1. Click the sub tag created above to go to `localhost:3000/tags/SUB_TAG` and click outside the navbar.
1. Check the browser console. It should be free of the error reported [here](https://github.com/reactioncommerce/reaction/issues/2626).
1. Repeat the last three steps a few more times using other sub-tags/tags.


<img width="464" alt="highlighted - screen shot 2017-09-06 at 6 42 33 pm" src="https://user-images.githubusercontent.com/26963575/30128058-2ed8f19c-9339-11e7-954b-12596b0e9180.png">
